receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -442,56 +437,63 @@ func (jr *jReceiver) buildProcessor(address string, factory apacheThrift.TProtoc
 }
 
 func (jr *jReceiver) startCollector(host receiver.Host) error {
-	tch, terr := tchannel.NewChannel("jaeger-collector", new(tchannel.ChannelOptions))
-	if terr != nil {
-		return fmt.Errorf("failed to create NewTChannel: %v", terr)
+	if !jr.collectorGRPCEnabled() && !jr.collectorHTTPEnabled() && !jr.collectorThriftEnabled() {
+		return nil
 	}
 
-	server := thrift.NewServer(tch)
-	server.Register(jaeger.NewTChanCollectorServer(jr.tchanServer))
+	if jr.collectorThriftEnabled() {
+		tch, terr := tchannel.NewChannel("jaeger-collector", new(tchannel.ChannelOptions))
+		if terr != nil {
+			return fmt.Errorf("failed to create NewTChannel: %v", terr)
+		}
 
-	taddr := jr.tchannelAddr()
-	tln, terr := net.Listen("tcp", taddr)
-	if terr != nil {
-		return fmt.Errorf("failed to bind to TChannel address %q: %v", taddr, terr)
-	}
-	tch.Serve(tln)
-	jr.tchanServer.tchannel = tch
-
-	// Now the collector that runs over HTTP
-	caddr := jr.collectorAddr()
-	cln, cerr := net.Listen("tcp", caddr)
-	if cerr != nil {
-		// Abort and close tch
-		tch.Close()
-		return fmt.Errorf("failed to bind to Collector address %q: %v", caddr, cerr)
-	}
+		server := thrift.NewServer(tch)
+		server.Register(jaeger.NewTChanCollectorServer(jr.tchanServer))
 
-	nr := mux.NewRouter()
-	apiHandler := app.NewAPIHandler(jr)
-	apiHandler.RegisterRoutes(nr)
-	jr.collectorServer = &http.Server{Handler: nr}
-	go func() {
-		_ = jr.collectorServer.Serve(cln)
-	}()
-
-	jr.grpc = grpc.NewServer(jr.config.CollectorGRPCOptions...)
-	gaddr := jr.grpcAddr()
-	gln, gerr := net.Listen("tcp", gaddr)
-	if gerr != nil {
-		// Abort and close tch, cln
-		tch.Close()
-		cln.Close()
-		return fmt.Errorf("failed to bind to gRPC address %q: %v", gaddr, gerr)
+		taddr := jr.collectorThriftAddr()
+		tln, terr := net.Listen("tcp", taddr)
+		if terr != nil {
+			return fmt.Errorf("failed to bind to TChannel address %q: %v", taddr, terr)
+		}
+		tch.Serve(tln)
+		jr.tchanServer.tchannel = tch
 	}
 
-	api_v2.RegisterCollectorServiceServer(jr.grpc, jr)
+	if jr.collectorHTTPEnabled() {
+		// Now the collector that runs over HTTP
+		caddr := jr.collectorHTTPAddr()
+		cln, cerr := net.Listen("tcp", caddr)
+		if cerr != nil {
+			// Abort and close tch
```

Comment:
How is `tch` closed?

Reply:
If `startCollector()` fails then `StartTraceReception()` calls `stopTraceReceptionLocked()`.  This will clean up tchannel or any other open services.

Removing the comment b/c it's misleading.